services/cars-service/src/main/java/za/co/protogen/core/impl/CarServiceImpl.java

```diff
@@ -9,32 +9,33 @@
 
 import java.util.Comparator;
 import java.util.List;
+
 @Service
 public class CarServiceImpl implements CarService {
 
-  private final CarRepository carRepository;
+    private final CarRepository carRepository;
 
     public CarServiceImpl(CarRepository carRepository) {
         this.carRepository = carRepository;
     }
 
     @Override
     public void addCar(Car car) {
-       carRepository.save(car);
-
+        carRepository.save(car);
     }
+
     @Override
     public void removeCar(String vin) {
 
-        if (!carRepository.existsById(Long.valueOf(vin))){
-           throw new IllegalStateException("car not found!!");
+        if (!carRepository.existsById(vin)) {
+            throw new IllegalStateException("car not found!!");
         }
-        carRepository.deleteById(Long.valueOf(vin));
+        carRepository.deleteById(vin);
     }
 
     @Override
     public Car getCarById(String vin) {
-        return carRepository.findById(Long.valueOf(vin)).orElse(null);
+        return carRepository.findById(vin).orElse(null);
```

Comment:
Here, create a Car object, then initialize it with the findById query.
Check if your object is null or not. If it's not null. Return else return a not found status.

Do a similar thing for other services